### ISSUE TITLE
Destination BigQuery: `404 Not found` table was not found in location `us-east1`

### ISSUE THREAD

**Issue by jrmanes** [airbytehq/airbyte#11914]
## Environment
- **Airbyte version**:  0.35.65-alpha (also tested: 0.35.56-alpha, 0.35.62-alpha)
- **OS Version / Instance**: Ubuntu - GCP Instance
- **Deployment**: docker-compose
- **Source Connector and version**: MySQL
- **Destination Connector and version**: BigQuery
- **Severity**:  High
- **Step where error happened**:  I can create the source and destination, the issue appears when I try to sync data

## Current Behavior
Related to the [issue](https://github.com/airbytehq/airbyte/issues/11875)
Once the application tries to insert data into the BigQuery, I got errors like this:
```
2022-04-12 03:18:38 normalization > 03:18:28.396010 [debug] [Thread-1  ]: BigQuery adapter: 404 Not found: Table <project>:<dataset>.<table> was not found in location us-east1
```
This is right, I have the tables created but with the following format:
```
<project>:<dataset>._airbyte_tmp_hqj_<table>
```
I thought the table will have to be created automatically, but this does not happen.

## Expected Behavior
Create the tables automatically and insert the data.

## Logs
```
2022-04-12 03:18:38 normalization > 03:18:28.396010 [debug] [Thread-1  ]: BigQuery adapter: 404 Not found: Table <project>:<dataset>.<table> was not found in location us-east1
```

## Are you willing to submit a PR?
Not yet

**Comment by woobione** [comment]
➕ 1️⃣ 

I've been ripping my hair out all day over this 😅 
We are moving our entire warehouse (BigQuery) from US to EU.
As a last step I deleted the airbyte-created datasets and proceeded to update our BigQuery connector to EU instead of US. It worked fine for 1 out of 3 connectors (Chartmogul) but not for MySQL (like the creator of this thread) and not for Chargebee.
After a lot of fiddling I decided to create a completely new BigQuery destination hoping it was maybe just some non-purged caching of region somewhere - and to my happiness it worked when I made a test of a single table from MySQL to BigQuery. However, when running our full database the same error occurred.

It seems to be happening in the transformation step.

Posting some error logs here:

This log makes me think there's something wrong in how the BigQuery connector decides the naming as it refers to `create or replace view data-analytics._airbyte_chargebee.subscription_stg` whilst it should be referring to `data-analytics.chargebee._airbyte_raw_subscription`. I've never seen a `_airbyte_chargebee` dataset created ever - everything happens under the primary dataset of that connector.

```
2022-07-15 14:19:55 [42mnormalization[0m >                                                             -----Query Job SQL Follows-----                                                             
2022-07-15 14:19:55 [42mnormalization[0m > 
2022-07-15 14:19:55 [42mnormalization[0m >     |    .    |    .    |    .    |    .    |    .    |    .    |    .    |    .    |    .    |    .    |    .    |    .    |    .    |    .    |
2022-07-15 14:19:55 [42mnormalization[0m >    1:/* {"app": "dbt", "dbt_version": "1.0.0", "profile_name": "normalize", "target_name": "prod", "node_id": "model.airbyte_utils.subscription_stg"} */
2022-07-15 14:19:55 [42mnormalization[0m >    2:
2022-07-15 14:19:55 [42mnormalization[0m >    3:
2022-07-15 14:19:55 [42mnormalization[0m >    4:  create or replace view `data-analytics`._airbyte_chargebee.`subscription_stg`
2022-07-15 14:19:55 [42mnormalization[0m >    5:  OPTIONS()
2022-07-15 14:19:55 [42mnormalization[0m >    6:  as 
2022-07-15 14:19:55 [42mnormalization[0m >    7:with __dbt__cte__subscription_ab1 as (
2022-07-15 14:19:55 [42mnormalization[0m >    8:
2022-07-15 14:19:55 [42mnormalization[0m >    9:-- SQL model to parse JSON blob stored in a single column and extract into separated field columns as described by the JSON Schema
2022-07-15 14:19:55 [42mnormalization[0m >   10:-- depends_on: `data-analytics`.chargebee._airbyte_raw_subscription
```


later this leads to this error:

```
2022-07-15 14:19:55 [42mnormalization[0m > 14:19:48.294928 [debug] [Thread-2  ]: Runtime Error in model subscription_stg (models/generated/airbyte_views/chargebee/subscription_stg.sql)
2022-07-15 14:19:55 [42mnormalization[0m >   404 Not found: Dataset data-analytics:_airbyte_chargebee was not found in location EU
2022-07-15 14:19:55 [42mnormalization[0m >   
2022-07-15 14:19:55 [42mnormalization[0m >   (job ID: e985f8ba-4177-4305-8659-1d16744dd03e)
```

Let me know if I can provide better information or logs somehow.
Also pasting our connector config below.

<img width="1279" alt="Skärmavbild 2022-07-15 kl  16 44 19" src="https://user-images.githubusercontent.com/673403/179247091-0fd87497-2d7e-46d0-9ec5-f6a1428fcb8b.png">
.

**Comment by woobione** [comment]
Okay! I've found what's causing (atleast my issue).
Once you've run a sync once in a location, say the US, hidden tables (not showing up in BigQuery's UI) are created prefixed with _airbyte_{schema} during normalization. If you then create another connector (or change the existing one) to another location, for example EU, it will try to create that same dataset, however, it can't because it already exists in the US.
I managed to solve this by using google cloud shell and running `bq ls --all` - seeing the hidden datasets. And then removing them with `bq rm _hidden_dataset_name`.
This should however be considered a bug, imo, since one might want to have two bigquery connectors writing to the same project but different locations - and this would be impossible today provided you want to use the same connectors.
A way to mitigate it would be to include location in these hidden tables - so for chargebee, for example, instead of `_airbyte_chargebee` it could be `_airbyte_chargebee_us` etc
